assets/js/invidious_iframe_api.js

```diff
@@ -0,0 +1,386 @@
+        search_params.append('enablejsapi','1');
+        search_params.append('widgetid',invidious_embed.widgetid);
+        this.widgetid = invidious_embed.widgetid;
+        invidious_embed.widgetid++;
+        if(autoplay){
+            search_params.append('autoplay',1);
+        }
+        else{
+            search_params.append('autoplay',0);
+        }
+        if(this.option_playerVars!==undefined){
+            for(var x in this.option_playerVars){
+                if(x!=='autoplay'&&x!=='start'&&x!=='end'){
+                    search_params.append(x,this.option_playerVars[x]);
+                }
+            }
+        }
+        if(startSeconds!==-1&&startSeconds>=0){
+            search_params.append('start',startSeconds);
+        }
+        if(endSeconds!==-1&&endSeconds>=0){
+            if(endSeconds>startSeconds){
+                search_params.append('end',endSeconds);
+            }
+            else{
+                throw 'invalid end seconds';
+            }
+        }
+        iframe_sorce += "?" + search_params.toString();
+        this.player_iframe.src = iframe_sorce;
+        if(autoplay){
+            this.player_status = 5;
+        }
+        this.eventdata = {};
+    }
+    loadVideoById(option,startSeconds){
+        this.playOtherVideoById(option,true,startSeconds);
+    }
+    cueVideoById(option,startSeconds){
+        this.playOtherVideoById(option,false,startSeconds);
+    }
+    cuevVideoByUrl(option,startSeconds){
+        this.playOtherVideoById(option,false,startSeconds);
+    }
+    loadVideoByUrl(option,startSeconds){
+        this.playOtherVideoById(option,true,startSeconds);
+    }
+    getDuration(){
+        return this.promise_send_event('getduration');
+    }
+    getVideoUrl(){
+        return this.target_origin + "/watch?v=" + this.videoId;
+    }
+    async getVideoEmbedCode(){
+        var title = await this.getVideoTitle();
+        return '<iframe width="560" height="315" src="' + this.target_origin + "/embed/" + this.videoId + '" title="' + title.replace('"','') + '" frameborder="0" allow="autoplay;encrypted-media;picture-in-picture; web-share" allowfullscreen></iframe>';
+    }
+    getCurrentTime(){
+        return this.promise_send_event('getcurrenttime');
+    }
+    constructor(element,options){
+        this.Player(element,options);
+        window.addEventListener('message',(ms)=>{this.receiveMessage(ms)});
+        this.message_wait = {getvolume:[],getmutestatus:[],getduration:[],getcurrenttime:[],getplaybackrate:[],getavailableplaybackrates:[],gettitle:[]};
+    }
+    async getVideoData(){
+        return {video_id:this.videoId,title:await this.promise_send_event('gettitle')};
+    }
+}
+function invidious_ready(func){
+    if(typeof func==='function'){
+        func();
+    }
+}
+const invidious = {Player:invidious_embed,PlayerState:{ENDED:0,PLAYING:1,PAUSED:2,BUFFERING:3,CUED:5},ready:invidious_ready};
+try{
+    onInvidiousIframeAPIReady();
+}
+catch{}
```

Comment:
Why empty `try-catch` expression? Let it broke loud and verbose
```suggestion
onInvidiousIframeAPIReady();
```

Reply:
When not to define function onInvidiousIframeAPIReady() at other javascript code ,it will be print error message.But original iframe api do not printerror message. That's why I use try-catch expression.But if you have a better way,please comment.

Reply:
Check whether onInvidiousIframeAPIReady define or not, and print error using console.error.